des.go

```diff
@@ -0,0 +1,82 @@
+//go:build linux && !cmd_go_bootstrap
+
+package openssl
+
+// #include "goopenssl.h"
+import "C"
+import (
+	"crypto/cipher"
+	"errors"
+	"runtime"
+)
+
+const desBlockSize = 8
+
+// SupportsDESCipher returns true if NewDESCipher is supported.
+func SupportsDESCipher() bool {
+	// True for stock OpenSSL 1.
+	// False for stock OpenSSL 3 unless the legacy provider is available.
+	return loadCipher(cipherDES, cipherModeECB) != nil
```

Comment:
It makes sense to me that `cipherModeECB` would always be the same as `cipherModeCBC`, but maybe worth pointing out in the comment so this doesn't look like it might be making a bad assumption?

(Also in `SupportsTripleDESCipher`)